llvm/lib/CheerpUtils/CheerpLowerAtomic.cpp

```diff
@@ -0,0 +1,26 @@
+#include "llvm/Cheerp/CheerpLowerAtomic.h"
+#include "llvm/Transforms/Scalar/LowerAtomicPass.h"
+
+using namespace llvm;
+using namespace cheerp;
+// Module pass that invokes the LLVM LowerAtomicPass on genericjs functions.
+PreservedAnalyses CheerpLowerAtomicPass::run(Module& M, ModuleAnalysisManager& MAM)
+{
+	FunctionAnalysisManager& FAM = MAM.getResult<FunctionAnalysisManagerModuleProxy>(M).getManager();
+	FunctionPassManager FPM;
+	FPM.addPass(LowerAtomicPass());
+
+	// Loop over the functions, and only pass genericjs ones to LowerAtomicPass
+	for (Function& F : M)
+	{
+		if (F.isDeclaration())
+			continue;
+
+		if (F.getSection() == "asmjs")
+			continue;
+
+		FPM.run(F, FAM);
+	}
+
+	return PreservedAnalyses::all();
```

Comment:
it's probably better to preserve none. We could do better and preserve analyses on functions that are not touched, but this whole pass is temporary so who cares.
But preserve all is probably incorrect

Reply:
Makes sense! I changed it to preserve none.